src/util/compat.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+export opaque type Compatible<T> = [Compat, T];
+type Compat = {|
+  +type: string,
+  +version: string,
+|};
+
+export function toCompat<T>(
+  type: string,
```

Comment:
A priori, I have a weak preference for taking the actual `compat: Compat` here instead of two unnamed arguments, but if you have a reason for doing it this way then I'm probably fine with it.

Reply:
Done